documentation/src/main/asciidoc/topics/proc_configuring_textbased_encoding.adoc

```diff
@@ -0,0 +1,28 @@
+[id='using_text_based_storage-{context}']
+= Using Text-Based Formats for Data
+You can configure {brandname} to encode data in text-based formats such as
+`text/plain`, `application/json`, or `application/xml`.
+
+.Procedure
+
+. Specify a text-based storage format as the MediaType for keys and values.
+. Optionally specify a character set such as `UTF-8`.
++
+The following example configures {brandname} to store entries with the `text/plain; charset=UTF-8` format:
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+include::config_examples/encoding_text_plain.xml[]
+----
++
+. Configure your clients.
+
+Hot Rod clients can use the
+
+* `org.infinispan.commons.marshall.StringMarshaller` handles plain text, JSON, XML, or any other text-based format.
+* ProtoStream can handle JSON only but offers more flexibility for storing data in binary format. However, you must create Serialization Contexts and provide Protobuf schemas (`.proto` files) to use ProtoStream.
+
```

Comment:
Some nuance here: for text based format, protostream can also be used without registering proto files :)